apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -429,6 +435,98 @@ func (s *StringTransform) Resolve(input interface{}) (interface{}, error) {
 	return fmt.Sprintf(s.Format, input), nil
 }
 
+// The list of supported ConvertTransform input and output types.
+const (
+	ConvertTransformTypeString  = "string"
+	ConvertTransformTypeBool    = "bool"
+	ConvertTransformTypeInt     = "int"
+	ConvertTransformTypeFloat64 = "float64"
+)
+
+var (
+	conversions = map[string]func(interface{}) (interface{}, error){
+		"string.string": func(i interface{}) (interface{}, error) {
+			return i, nil
+		},
```

Comment:
Could we detect and handle the no-op case in which the from and to types are the same down in `Resolve`, so that we don't need to include the handful of no-op conversions in this map?

Reply:
yeah, just added that.